test/e2e/test_e2e.py

```diff
@@ -0,0 +1,717 @@
+
+    retries = 5
+    for i in range(retries):
+        print("Starting MRD Storage Server on port", port)
+        proc = subprocess.Popen(["mrd-storage-server", "--require-parent-pid", str(os.getpid())],
+                                stdout=log,
+                                stderr=log,
+                                env=storage_server_environment)
+
+        try:
+            wait_for_storage_server(port, proc)
+            return proc
+        except:
+            # If the process has exited, it might be because the
+            # port was in use. This can be because the previous storage server
+            # instance was just killed. So we try again.
+            if proc.poll() is not None and i < retries:
+                time.sleep(1)
+            else:
+                proc.kill()
+                raise
+
+def start_gadgetron_instance(*, log_stdout, log_stderr, port, storage_address, env=environment):
+    print("Starting Gadgetron instance on port", port)
+    proc = subprocess.Popen(["gadgetron", "-p", port, "-E", storage_address],
+                            stdout=log_stdout,
+                            stderr=log_stderr,
+                            env=env)
+    return proc
+
+
+@pytest.fixture(autouse="true")
+def start_storage(external, storage_port, tmp_path: Path):
+    storage_path = os.path.join(tmp_path, "storage")
+    os.mkdir(storage_path)
+
+    if not external:
+        print("Starting storage server on port", storage_port)
+        with open(os.path.join(tmp_path, 'storage.log'), 'w') as log:
+            storage = start_storage_server(log=log,
+                        port=str(storage_port),
+                        storage_folder=str(storage_path))
+            
+            yield
+
+            storage.kill()
+
+    else:
+        return
+
+@pytest.fixture
+def start_gadgetron_sever(port, external, storage_port, tmp_path):
+    instance = None
+    def _start_gadgetron():
+        nonlocal instance
+
+        storage_address = "http://localhost:" + storage_port
+
+        if not external:
+            print("Starting Gadgetron instance on port {} with logs in {}".format(port, tmp_path))
+
+            with open(os.path.join(tmp_path, 'gadgetron.log.out'), 'w') as log_stdout:
+                with open(os.path.join(tmp_path, 'gadgetron.log.err'), 'w') as log_stderr:
+                    instance = start_gadgetron_instance(log_stdout=log_stdout, log_stderr=log_stderr, port=port, storage_address=storage_address)
+
+    yield _start_gadgetron
+
+    if instance != None:
+        instance.kill()
+
+
+@pytest.fixture
+def start_gadgetron_sever_with_additional_nodes(tmp_path: Path, port, storage_port, external):
+    instances = []
+
+    def _start_gadgetron_with_additional_nodes(fileConfig:dict):
+        nonlocal instances
+
+        if external:
+            return
+
+        base_port = 9050
+        number_of_nodes = 2
+
+        if 'config' in fileConfig:
```

Comment:
Is there a 'config' section in fileConfig, how does this work?

Reply:
I think when running distributed tests, we should check that work is sent to the worker nodes (there should be some indication in their stderr).

Reply:
That should have been 'distributed' not 'config', that has been fixed, and I'm looking into how to test that the work was sent to the worker nodes.